crates/l2/common/src/l1_messages.rs

```diff
@@ -0,0 +1,112 @@
+use std::sync::LazyLock;
+
+use ethereum_types::{Address, H256};
+use ethrex_common::{
+    H160,
+    types::{Receipt, Transaction},
+};
+use keccak_hash::keccak;
+
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+pub const L1MESSENGER_ADDRESS: Address = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0xff, 0xfe,
+]);
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+/// Represents a message from the L2 to the L1
+pub struct L1Message {
+    /// Address that called the L1Messanger
+    pub from: Address,
+    /// Hash of the data given to the L1Messanger
+    pub data_hash: H256,
+    /// L2 Transaction the message was included in, for ease of usage
+    pub tx_hash: H256,
```

Comment:
Let's put this field first, just to have them in the same order they are encoded in